drivers/timer/xlnx_psttc_timer.c

```diff
@@ -1,187 +1,186 @@
-#error ("No timer is specified")
-#endif
+#define TICKS_PER_SEC		CONFIG_SYS_CLOCK_TICKS_PER_SEC
+#define CYCLES_PER_SEC		TIMER_CLOCK_FREQUECY
+#define CYCLES_PER_TICK		(CYCLES_PER_SEC / TICKS_PER_SEC)
+#define CYCLES_NEXT_MIN		(10000)
+#define CYCLES_NEXT_MAX		(4294967295)
 
-#define XTTCPS_CLK_CNTRL_OFFSET	   0x00000000U  /**< Clock Control Register */
-#define XTTCPS_CNT_CNTRL_OFFSET	   0x0000000CU  /**< Counter Control Register*/
-#define XTTCPS_COUNT_VALUE_OFFSET  0x00000018U  /**< Current Counter Value */
-#define XTTCPS_INTERVAL_VAL_OFFSET 0x00000024U  /**< Interval Count Value */
-#define XTTCPS_MATCH_0_OFFSET	   0x00000030U  /**< Match 1 value */
-#define XTTCPS_MATCH_1_OFFSET	   0x0000003CU  /**< Match 2 value */
-#define XTTCPS_MATCH_2_OFFSET	   0x00000048U  /**< Match 3 value */
-#define XTTCPS_ISR_OFFSET	   0x00000054U  /**< Interrupt Status Register */
-#define XTTCPS_IER_OFFSET	   0x00000060U  /**< Interrupt Enable Register */
-
-/* Clock Control Register definitions */
-#define XTTCPS_CLK_CNTRL_PS_EN_MASK	0x00000001U  /**< Prescale enable */
-#define XTTCPS_CLK_CNTRL_PS_VAL_MASK	0x0000001EU  /**< Prescale value */
-#define XTTCPS_CLK_CNTRL_PS_VAL_SHIFT	1U  /**< Prescale shift */
-#define XTTCPS_CLK_CNTRL_PS_DISABLE	16U  /**< Prescale disable */
-#define XTTCPS_CLK_CNTRL_SRC_MASK	0x00000020U  /**< Clock source */
-#define XTTCPS_CLK_CNTRL_EXT_EDGE_MASK	0x00000040U  /**< External Clock edge */
-
-/* Counter Control Register definitions */
-#define XTTCPS_CNT_CNTRL_DIS_MASK	0x00000001U /**< Disable the counter */
-#define XTTCPS_CNT_CNTRL_INT_MASK	0x00000002U /**< Interval mode */
-#define XTTCPS_CNT_CNTRL_DECR_MASK	0x00000004U /**< Decrement mode */
-#define XTTCPS_CNT_CNTRL_MATCH_MASK	0x00000008U /**< Match mode */
-#define XTTCPS_CNT_CNTRL_RST_MASK	0x00000010U /**< Reset counter */
-#define XTTCPS_CNT_CNTRL_EN_WAVE_MASK	0x00000020U /**< Enable waveform */
-#define XTTCPS_CNT_CNTRL_POL_WAVE_MASK	0x00000040U /**< Waveform polarity */
-#define XTTCPS_CNT_CNTRL_RESET_VALUE	0x00000021U /**< Reset value */
-
-/* Interrupt register masks */
-#define XTTCPS_IXR_INTERVAL_MASK	0x00000001U  /**< Interval Interrupt */
-#define XTTCPS_IXR_MATCH_0_MASK		0x00000002U  /**< Match 1 Interrupt */
-#define XTTCPS_IXR_MATCH_1_MASK		0x00000004U  /**< Match 2 Interrupt */
-#define XTTCPS_IXR_MATCH_2_MASK		0x00000008U  /**< Match 3 Interrupt */
-#define XTTCPS_IXR_CNT_OVR_MASK		0x00000010U  /**< Counter Overflow */
-#define XTTCPS_IXR_ALL_MASK		0x0000001FU  /**< All valid Interrupts */
-
-#define XTTC_MAX_INTERVAL_COUNT 0xFFFFFFFFU /**< Maximum value of interval counter */
-
-static u32_t accumulated_cycles;
-static s32_t _sys_idle_elapsed_ticks = 1;
-
-static int xttc_calculate_interval(u32_t *interval, u8_t *prescaler)
-{
-	u32_t tmpinterval = 0;
-	u8_t tmpprescaler = 0;
-	unsigned int tmpval;
+BUILD_ASSERT_MSG(CYCLES_PER_SEC == CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,
+		 "Configured system timer frequency does not match the TTC "
+		 "clock frequency");
```

Comment:
Better to check this directly between the DTS and the kconfig values.  Basically the same comment as above; having a local CYCLES_PER_SEC obscures what's happening here (i.e. that you're validating our two configuration mechanisms aren't lying to you).

Reply:
I will replace `CYCLES_PER_SEC` here with `TIMER_DT(XLNX_TTCPS_CLOCK_FREQUENCY)`.

Directly using the DT symbol is not an option because we need to map the TTC instance number before doing so.